### DISCUSSION TITLE
Warn about missing curl extension?

### DISCUSSION THREAD

**Discussion by woganmay** [composer/composer#9558] [General]
I've been setting up a new laptop for Laravel development work, and installed Composer using the script on the site. I then tried to create a new laravel/laravel project, and it came back with:

    Memory usage: 28.83MiB (peak: 163.46MiB), time: 403.45s

It took ages but ran with no complaints. Only after I installed the `php7.4-curl` extension:

    Memory usage: 31.44MiB (peak: 166.73MiB), time: 25.26s

I think it's really cool that Composer will find a way to work even without that extension installed (guessing lots of file_get_contents?), but installing it cut 90% off my install time. Would it be worth adding some sort of notice (not a warning) that just suggests installing the curl extension to improve performance?

**Comment by stof** [comment]
`composer diagnose` already displays such warning

**Comment by Seldaek** [comment]
I think it may make sense to warn here.. Because the difference can be quite big and it seems ~97% have curl enabled anyway so the amount of people seeing a warning will be small.

**Comment by Seldaek** [comment]
https://github.com/composer/composer/commit/4a740e3e61643900291fa5c34b76f76c662b06e9 fixes this.

**Reply by woganmay** [reply]
Thank you!
